src/modals/AnonymizedDataConsentScreen.spec.tsx

```diff
@@ -5,125 +5,104 @@ import { useNavigation } from "@react-navigation/native"
 import AnonymizedDataConsentScreen from "./AnonymizedDataConsentScreen"
 import { factories } from "../factories"
 import { AnalyticsContext } from "../AnalyticsContext"
-import { Stacks } from "../navigation"
 
 jest.mock("@react-navigation/native")
-describe("AnonymizedDataConsentScreen", () => {
-  describe("when a user clicks the close button", () => {
-    it("navigates back to settings without changing their consent", () => {
-      const navigationSpy = jest.fn()
-      ;(useNavigation as jest.Mock).mockReturnValue({ goBack: navigationSpy })
-
-      const updateUserConsent = jest.fn()
-      const context = factories.analyticsContext.build({
-        userConsentedToAnalytics: false,
-        updateUserConsent,
-      })
-      const { getByTestId } = render(
-        <AnalyticsContext.Provider value={context}>
-          <AnonymizedDataConsentScreen />
-        </AnalyticsContext.Provider>,
-      )
-
-      const closeButton = getByTestId("close-consent-screen")
-      fireEvent.press(closeButton)
-
-      expect(navigationSpy).toHaveBeenCalled()
-      expect(updateUserConsent).not.toHaveBeenCalled()
-    })
-  })
 
+describe("AnonymizedDataConsentScreen", () => {
   describe("when a user has consented to data sharing", () => {
     it("displays the correct status", () => {
       const context = factories.analyticsContext.build({
         userConsentedToAnalytics: true,
       })
-      const { queryByText } = render(
+      ;(useNavigation as jest.Mock).mockReturnValue({
+        setOptions: jest.fn(),
+      })
+      const { getByText } = render(
         <AnalyticsContext.Provider value={context}>
           <AnonymizedDataConsentScreen />
         </AnalyticsContext.Provider>,
       )
 
-      const headerText = queryByText("You are sharing anonymized data")
-      const buttonText = queryByText("Stop Sharing Data")
+      const buttonText = getByText("Stop Sharing Data")
 
-      expect(headerText).not.toBeNull()
-      expect(buttonText).not.toBeNull()
+      expect(buttonText).toBeDefined()
     })
 
     describe("and they press to revoke consent", () => {
-      it("updates their consent status and navigates to settings", async () => {
+      it("updates their consent status and navigates back", async () => {
         expect.assertions(2)
         const navigationSpy = jest.fn()
         ;(useNavigation as jest.Mock).mockReturnValue({
-          navigate: navigationSpy,
+          goBack: navigationSpy,
+          setOptions: jest.fn(),
```

Comment:
@aledustet Is there a better way to mock `setOptions`? I couldn't figure out how to do it any other way than just mocking it in each describe block.